app/src/main/java/org/onionshare/android/ui/MainActivity.kt

```diff
@@ -55,9 +68,13 @@ class MainActivity : ComponentActivity() {
 
     private fun onFabClicked() {
         try {
-            contentLauncher.launch("*/*")
+            contentLauncher.launch(arrayOf("*/*"))
         } catch (e: ActivityNotFoundException) {
-            Toast.makeText(this, R.string.add_files_not_supported, LENGTH_SHORT).show()
+            try {
+                contentFallbackLauncher.launch("*/*")
+            } catch (e: ActivityNotFoundException) {
+                Toast.makeText(this, R.string.add_files_not_supported, LENGTH_SHORT).show()
```

Comment:
Newbie Android developer: believes the docs.

Experienced Android developer: does what the docs say but uses a try/catch block just in case.

Expert Android developer: uses nested try/catch blocks with different approaches and still expects that it won't work on all devices.